app/code/Magento/Customer/Model/Plugin/CustomerFlushFormKey.php

```diff
@@ -45,7 +45,7 @@ public function aroundExecute(FlushFormKey $subject, callable $proceed, ...$args
         $currentFormKey = $this->dataFormKey->getFormKey();
         $proceed(...$args);
         $beforeParams = $this->session->getBeforeRequestParams();
-        if ($beforeParams['form_key'] == $currentFormKey) {
+        if (isset($beforeParams['form_key']) && $beforeParams['form_key'] == $currentFormKey) {
```

Comment:
Can you please switch it to strict comparison?
`$beforeParams['form_key'] === $currentFormKey`